### PR TITLE
Add Jakarta EE compatible Socket Mode client ref: #919

### PR DESCRIPTION
**Pull request by seratch**
This pull request adds two new optional modules:
* slack-jakarta-socket-mode-client
* bolt-jakarta-socket-mode

The currently available Socket Mode client's default implementation uses tyrus-standalone-client **1.x**, which is compatible with **javax.websocket-api** APIs. The Jakarta EE version of this interface is the **jakarta.websocket-client-api** APIs, and tyrus-standalone-client **2.x** is compatible with it. Since it's not feasible to have both tyrus-standalone-client 1.x and 2.x in the same module's dependencies, I have added a new module named slack-jakarta-socket-mode-client. See https://github.com/slackapi/java-slack-sdk/issues/919 for more details.

Developers can initialize the Jakarta-compatible SocketModeClient this way:

```java
import com.slack.api.Slack;
import com.slack.api.jakarta_socket_mode.JakartaSocketModeClientFactory;

public class Example {
  public static void main(String[] args) throws Exception {
    var appToken = System.getenv("SLACK_APP_TOKEN");
    var slack = Slack.getInstance();
    // Java EE compatible Socket Mode client
    slack.socketMode(appToken).connect();
    // Jakarta EE compatible Socket Mode client
    JakartaSocketModeClientFactory.create(slack, appToken).connect();
  }
}
```

In the same way, I’ve added a new Jakarta-compatible module, which is equivalent to bolt-socket-mode. Here is the demo code. As you can see, just replace the dependency and imports in the code:

```java
import com.slack.api.bolt.App;
import com.slack.api.bolt.jakarta_socket_mode.SocketModeApp;

public class Example {
  public static void main(String[] args) throws Exception {
    var app = new App();
    app.command("/hi", (req, ctx) -> {
      ctx.say("Hi there!");
      return ctx.ack();
    });
    var appToken = System.getenv("SLACK_APP_TOKEN");
    // Switch from com.slack.api.bolt.socket_mode to com.slack.api.bolt.jakarta_socket_mode
    new SocketModeApp(appToken, app).start();
  }
}
```

The reason behind this enhancement is that many Java-focused companies are planning to eliminate the legacy javax.* dependencies from their project settings. I don't think the short-term risk of having a javax.websocket dependency is significant, but it seems it's about time to provide an option for migration on the developers' side.

### Category (place an `x` in each of the `[ ]`)

* [ ] **bolt** (Bolt for Java)
* [x] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [x] **slack-app-backend** (The primitive layer of Bolt for Java)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you agree to those rules.

### CODE REVIEW

File: slack-jakarta-socket-mode-client/src/main/java/com/slack/api/jakarta_socket_mode/impl/JakartaSocketModeClientTyrusImpl.java [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Socket Mode Client
```
Comment (WilliamBergamin) [inline slack-jakarta-socket-mode-client/src/main/java/com/slack/api/jakarta_socket_mode/impl/JakartaSocketModeClientTyrusImpl.java:36]:
```suggestion
 * Jakarta Socket Mode Client
```

File: slack-jakarta-socket-mode-client/src/main/java/com/slack/api/jakarta_socket_mode/impl/JakartaSocketModeClientTyrusImpl.java [hunk 2]
```diff
@@ -0,0 +1,435 @@
+package com.slack.api.jakarta_socket_mode.impl;
```
Comment (WilliamBergamin) [inline slack-jakarta-socket-mode-client/src/main/java/com/slack/api/jakarta_socket_mode/impl/JakartaSocketModeClientTyrusImpl.java:1]: I see this is almost an exact copy of `SocketModeClientTyrusImpl.java`, which is production worthy 🧠

**Review by WilliamBergamin — APPROVED** [review]
This looks good to me 💯  the changes seem to be mainly be a copy of existing code 👍 

I'm assuming this code has been stable for a while, maintaining this type of duplication should not be a problem

File: slack-jakarta-socket-mode-client/src/main/java/com/slack/api/jakarta_socket_mode/impl/JakartaSocketModeClientTyrusImpl.java (hunk 2 above)
Comment (seratch) [reply slack-jakarta-socket-mode-client/src/main/java/com/slack/api/jakarta_socket_mode/impl/JakartaSocketModeClientTyrusImpl.java:1]: Yeah, you're right!

**Pull request merged**